tutorials/roofit/rf317_global_observables.C

```diff
@@ -0,0 +1,169 @@
+/// \file
+/// \ingroup tutorial_roofit
+/// \notebook -js
+/// This tutorial explains the concept of global observables in RooFit, and
+/// showcases how their values can be stored either in the model or in the
+/// dataset.
+///
+/// \macro_code
+///
+/// \date January 2022
+/// \author Jonas Rembser
+
+#include <RooArgSet.h>
+#include <RooConstVar.h>
+#include <RooDataSet.h>
+#include <RooFitResult.h>
+#include <RooGaussian.h>
+#include <RooProdPdf.h>
+#include <RooRealVar.h>
+
+// Helper function to call RooAbsPdf::fitTo.
+template <class... Args_t>
+std::unique_ptr<RooFitResult> doFit(RooAbsPdf &model, RooAbsData &data, Args_t &&...args)
+{
+   using namespace RooFit;
+
+   // we clone the model to not change the parameters by fitting
+   std::unique_ptr<RooAbsPdf> modelClone{static_cast<RooAbsPdf *>(model.cloneTree())};
+
+   return std::unique_ptr<RooFitResult>{modelClone->fitTo(data, PrintLevel(-1), Save(), std::forward<Args_t>(args)...)};
+}
+
+void rf317_global_observables()
+{
+   using namespace RooFit;
+
+   // Introduction
+   // ------------
+   //
+   // With RooFit, you usually optimize some model parameters `p` to minimize
+   // the negative log-likelihood `l` given the per-event or per-bin
+   // observations `x`:
+   //
+   // l( x | p )
+   //
+   // Often, the parameters are constrained with some prior negative-log
+   // likelihood `c`, which doesn't depend on the observables `x`:
+   //
+   // l'( x | p ) = l( x | p ) + c( p )
+   //
+   // Usually, these constraint term depend on some auxiliary measurements of
+   // other observables `g`. The constraint term is then the likelihood of the
+   // so-called global observables:
+   //
+   // l'( x | p ) = l( x | p ) + c( g | p )
+   //
+   // For example, think of a model where the true luminosity `lumi` is a
+   // nuisance parameter that is constrained by an auxiliary measurement
+   // `lumi_obs` with uncertainty `lumi_obs_sigma`:
+   //
+   // l'(data | mu, lumi) = l(data | mu, lumi) * Gauss(lumi | lumi_obs, lumi_obs_sigma)
+   //
+   // Since a Gaussian is symmetric in `lumi` and `lumi_obs`, we can
+   // reformulate the likelihood to make it apparent that `lumi_obs` is a
+   // global observables:
+   //
+   // l'(data | mu, lumi) = l(data | mu, lumi) * Gauss(lumi_obs | lumi, lumi_obs_sigma)
+   //
+   // Note: be extra careful to define global observables and constrained
+   // parameters correctly in the case of non-symmetric constraints pdfs! For
+   // example, if the constraint would be a Poissonian, the global observable
+   // really has to be the observable of the Poissonian:
+   //
+   // l'(data | mu, lumi) = l(data | mu, lumi) * Poisson(lumi_obs | lumi)
```

Comment:
Of course, because that's how you do it?? Remove?

Or better:
Don't mention the reformulation and do something like:
> If you wanted to constrain event counts, you would e.g. use: `Pois( count_obs | count)`

Don't use lumi. Poisson constraints don't make sense here.

Reply:
I understand where you're coming from, indeed it's a bit awkward how I first define the constraint term the other way around. But that's because in the tutorial where constraints are introduced it's also the other way around, so I wanted to present both ways as equally valid for symmetric constraints.
https://github.com/root-project/root/blob/master/tutorials/roofit/rf604_constraints.C#L53

But probably it's best to just update rf604 and then be more assertive in this tutorial that the global observable as the observable of the pdf is the correct convention!